### PR TITLE
storage: Add system.dt_local_indexes

### PR DESCRIPTION
**Pull request by Lloyd-Pottiger**
### What problem does this PR solve?

Issue Number: ref #9032

Problem Summary:

### What is changed and how it works?

Pick https://github.com/tidbcloud/tiflash-cse/pull/211

```commit-message
storage: Add system.dt_local_indexes
```

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

File: dbms/src/Storages/DeltaMerge/DeltaMergeStore.h [hunk 1]
```diff
@@ -173,6 +174,19 @@ struct StoreStats
...
 };
 
+struct LocalIndexStats
+{
+    String column_name{};
+    UInt64 column_id{};
+    String index_kind{};
+
+    UInt64 rows_stable_indexed{}; // Total rows
+    UInt64 rows_stable_not_indexed{}; // Total rows
```
Comment (CalvinNeo) [inline dbms/src/Storages/DeltaMerge/DeltaMergeStore.h:184]: What's these comments?

**Review by CalvinNeo — APPROVED** [review]
lgtm

**Review by breezewish — APPROVED** [review]

File: dbms/src/Storages/DeltaMerge/DeltaMergeStore.h (hunk 1 above)
Comment (Lloyd-Pottiger) [reply dbms/src/Storages/DeltaMerge/DeltaMergeStore.h:184]: total rows of all segments.

**Pull request merged**